### ISSUE TITLE
A few issues with `bit_vector`

### ISSUE THREAD

**Issue by HolyBlackCat** [bkille/BitLib#39]
Pardon me for dumping several issues into a single ticket.

- [x] `bit_vector` lacks a move constructor.
- [x] The copy assignment should take the parameter by const reference, otherwise this fails:
    ```cpp
    const bit_vector<unsigned int> foo();
    bit_vector<unsigned int> v;
    v = foo();
    ```
- [x] `digits` should be `static constexpr`. There's no reason to store it in each vector instance.
- [x] Move constructor and assignment need to be `noexcept`, otherwise when placed in a standard container such as `std::vector`, they'll be copied on resize instead of being moved; see [How to enforce move semantics when a vector grows?](https://stackoverflow.com/questions/8001823/how-to-enforce-move-semantics-when-a-vector-grows)
- [x] The move assignment leaves the source vector in an inconsistent state, since its vector is now empty but `length_` is non-zero. I would zero it. Same applies to the future move constructor.
    On a side note, the copy constructor an assignment can (and should) be `= default;`ed.
- [x] Using `std::ceil` with floating-point division looks odd. It'll give inaccurate results at large sizes, and I expect it to be slower than the integral equivalent: `(a + b - 1) / b`.
- [x] `bit_vector::bit_vector(std::string_view s)` does something weird. It interprets the string as bytes, and takes the LSB of each character. It probably should reject everything except `'0'` and `'1'`, and there should be a separate constructor with a pair of `bool` iterators.

The rest is purely nitpicking:
- [x] `constexpr` is missing on a few functions: non-const `operator[]` and the assignment operators.
- [x] Compiling with `-Wextra-semi` would give a bunch of warnings on `;`s after closing braces.

**Comment by bkille** [comment]
Wow, thanks for the detailed feedback! All great points I'll address them ASAP

**Comment by bkille** [comment]
@HolyBlackCat I think I've addressed all of these in my most recent PR #41  (with the `default`s being added directly to master). Please re-open if you see anything amiss and again, thank you for reviewing this!
